shiny/ui/_chat.py

```diff
@@ -584,16 +584,20 @@ async def _handle_error():
 
     async def _append_message_stream(self, message: AsyncIterable[Any]):
         id = _utils.private_random_id()
-
-        empty = ChatMessage(content="", role="assistant")
-        await self._append_message(empty, chunk="start", stream_id=id)
-
+        is_start = True
         try:
             async for msg in message:
-                await self._append_message(msg, chunk=True, stream_id=id)
+                await self._append_message(
+                    msg, chunk="start" if is_start else True, stream_id=id
+                )
+                is_start = False
```

Comment:
Turns out this change will change the order of messages in something like this:

```python
from shiny import reactive
from shiny.express import ui

ui.page_opts(title="Hello Chat")

chat = ui.Chat(id="chat")
chat.ui()

@reactive.effect
async def _():
    await chat.append_message_stream(("FIRST ", "FIRST ", "FIRST"))

@reactive.effect
async def _():
    await chat.append_message("SECOND SECOND SECOND")
```

This seems important enough that I'd like to try to keep synchronously starting the stream, if possible. Seems we could keep that and achieve a better/simpler loading message experience -- I have a look into it.

Reply:
Yeah, sounds good.

Reply:
I'm not super sympathetic to concurrent operations on the chat object like this, but I also don't love that (with my code) the front end doesn't know that message streaming is even happening until the first chunk is received.